core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -199,8 +199,8 @@ private[spark] object ReliableCheckpointRDD extends Logging {
 
     val finalOutputName = ReliableCheckpointRDD.checkpointFileName(ctx.partitionId())
     val finalOutputPath = new Path(outputDir, finalOutputName)
-    val tempOutputPath =
-      new Path(outputDir, s".$finalOutputName-attempt-${ctx.attemptNumber()}")
+    val tempOutputPath = new Path(outputDir,
+      s".$finalOutputName-attempt-${ctx.stageAttemptNumber()}-${ctx.attemptNumber()}")
```

Comment:
`stageAttemptNumber` isn't unique among stages and `attemptNumber` isn't unique among tasks within the same stage. So it seems that this could still lead to the file name conflict. e.g. task 0.0 from stage 0.0 could conflict with task 1.0 from stage 1.0  (different stage) and task 1.0 from stage 1.0 could conflict with task 2.0 from stage 1.0 (same stage).

I think the unique file format should be `...-stageId-stageAttemptId-taskId-taskAttemptId-...`.

Reply:
A RDD can across stages?

Reply:
No, but all attempts Id starts from 0.

Reply:
Oh, I see we suffix the checkpoint path with rdd id...

Reply:
If we just want a unique file name, can we use the task id? It's unique within the Spark application.

Reply:
Do you mean `taskAttemptId`?

Reply:
Yes, `taskAttemptId`, we also use it in the shuffle map file for making the file name unique. https://github.com/apache/spark/pull/24892#issuecomment-526215837

Reply:
OK. Let me create a follow-up for it. Thanks.